docs/getting-started/quickstart-programming.md

```diff
@@ -286,12 +316,11 @@ You should see an output like:
 [info]: Greeting to Mike recorded in the database!
 ```
 
-Notice how DBOS automatically resumes your workflow from where it left off.
-It doesn't re-send the greeting email, but does record the previously-sent greeting in the database.
-This reliability is a core feature of DBOS: workflows always run to completion and each of their operations executes exactly once.
-To learn more about workflows, check out our [tutorial](../tutorials/workflow-tutorial.md) and [explainer](../explanations/how-workflows-work.md).
+Notice how DBOS automatically resumes your workflow from where it left off. It does not sign the guestbook twice but records the outstanding greeting in the database. This reliability is a core feature of DBOS: workflows always run to completion and each of their operations executes exactly once. To learn more about workflows, check out our [tutorial](../tutorials/workflow-tutorial.md) and [explainer](../explanations/how-workflows-work.md).
+
+Note that we start the workflow asynchronously using `startWorkflow` in the `GreetingWorkflow` invocation. This returns the response to the caller before the workflow has a chance to finish. The caller can then check on workflow completion after the fact, by calling `/greetings` for example. This behavior is preferred when the caller expects a fast response, such as with a [payment webhook](https://www.dbos.dev/blog/open-source-typescript-stripe-processing). You can change this switching `startWorkflow` to `invokeWorkflow`. You can then edit `GreetingWorkflow` to return a value that `Greeting` passes back to the client.
```

Comment:
This paragraph can be clearer, emphasizing we return after we've marked the workflow has been started.
```suggestion
Note that we start the workflow asynchronously using `startWorkflow` in the `GreetingWorkflow` invocation. This returns the response to the caller as soon as the workflow starts, without waiting for the workflow to return. DBOS guarantees that the workflow continues to process to completion asynchronously. This behavior is preferred when the caller expects a fast response, such as with a [payment webhook](https://www.dbos.dev/blog/open-source-typescript-stripe-processing). You can change this switching `startWorkflow` to `invokeWorkflow`. You can then edit `GreetingWorkflow` to return a value that `Greeting` passes back to the client.
```